cns/restserver/api.go

```diff
@@ -1029,7 +1031,27 @@ func (service *HTTPRestService) unpublishNetworkContainer(w http.ResponseWriter,
 
 	ctx := r.Context()
 
-	if !service.isNetworkJoined(req.NetworkID) {
+	var unpublishBody nmagent.DeleteContainerRequest
+	var azrNC bool
+	err = json.Unmarshal(req.DeleteNetworkContainerRequestBody, &unpublishBody)
+	if err != nil {
+		// If the body contains only `""\n`, it is non-AZR NC
+		// In this case, we should not return an error
+		// However, if the body is not `""\n`, it is invalid and therefore, we must return an error
+		if !bytes.Equal(req.DeleteNetworkContainerRequestBody, []byte(`""`+"\n")) {
```

Comment:
the bytes are hard to parse due to the mix of raw and regular string literals. Easier to just write `[]byte{34, 34, 10}` and explain what it is in the comment?

Reply:
Yeah, i had a tossup between the two and thought visually the bytes make more sense, but I can change that up.

Reply:
Changed